### PR TITLE
Issue/981 Added publisher name acronyms to elasticsearch

### PR DESCRIPTION
**Pull request by t83714**
### What this PR does

Added publisher name acronyms to `elasticsearch` so that publisher acronym will be recognised by search API.

Please note: keywords `AND` , `THE` & `OF` will be excluded from making acronyms.

e.g. Australian Pesticides and Veterinary Medicines Authority will become `APVMA` rather than `APAVMA`

Some example acronyms for testing:
- APVMA
  - Australian Pesticides and Veterinary Medicines Authority
- ACBPS
  - Australian Customs and Border Protection Service
- DEE
  - Department of the Environment and Energy

### Checklist
- [ ] There are unit tests to verify my changes are correct
- [x] I've updated CHANGES.md with what I changed.
- [x] I've linked this PR to an issue in ZenHub (or there is no issue) and dragged it to the _Pull Request_ column

### CODE REVIEW

File: magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala [hunk 1]
```diff
@@ -171,11 +171,18 @@ object Registry {
 
   trait RegistryConverters extends RegistryProtocols {
 
+    def getAcronymFromPublisherName(publisherName:Option[String]): Option[String] = {
+      publisherName
+        .map("""[^a-zA-Z\s]""".r.replaceAllIn(_,""))
+        .map("""\s""".r.split(_).map(_.trim.toUpperCase).filter(!List("","AND","THE").contains(_)).map(_.take(1)).mkString)
```
Comment (AlexGilleran) [inline magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala]: "of" too I would think

File: magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/Queries.scala [hunk 2]
```diff
@@ -22,7 +22,9 @@ object Queries {
   def publisherQuery(strategy: SearchStrategy)(publisher: FilterValue[String]) = {
     handleFilterValue(publisher, (publisherString: String) =>
       strategy match {
-        case SearchStrategy.MatchAll => matchQuery("publisher.name.keyword_lowercase", publisherString)
+        case SearchStrategy.MatchAll => multiMatchQuery(publisherString)
+          .fields("publisher.acronym", "publisher.name.keyword_lowercase")
+          .minimumShouldMatch("-50%")
         case SearchStrategy.MatchPart =>
           multiMatchQuery(publisherString)
             .fields("publisher.name", "publisher.name.english")
```
Comment (AlexGilleran) [inline magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/Queries.scala]: should probably put the acronym here too :)

File: magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala (hunk 1 above)
Comment (t83714) [reply magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala]: We probably want to keep `OF` in order to be consistent with other search engines (e.g. google).
Tested `Department of the Environment and Energy` on google:
So, if you search  `DOEE` on google, you will get `Department of the Environment and Energy` as the first result.
But if you search `DEE`, you won't get `Department of the Environment and Energy` related result.

File: magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/Queries.scala (hunk 2 above)
Comment (t83714) [reply magda-search-api/src/main/scala/au/csiro/data61/magda/search/elasticsearch/Queries.scala]: Thanks a lot!
Have corrected this~

File: magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala (hunk 1 above)
Comment (AlexGilleran) [reply magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala]: Yeah but the convention people will be used to is dropping the of - e.g. the Department of Home Affairs is known as DHA, not DOHA, Department of Foreign Affairs and Trade is known as DFAT etc.

Comment (t83714) [reply magda-scala-common/src/main/scala/au/csiro/data61/magda/model/Registry.scala]: Make sense~ will updated the list~

**Comment by AlexGilleran** [comment]
Looks fine but still needs unit tests for acronym generation - I'll merge this for 0.0.40 but leave the task in In Progress until there's tests :).

**Pull request merged**